src/qtfirebaseadmob.cpp

```diff
@@ -559,6 +580,46 @@ void QtFirebaseAdMobBase::init()
     }
 }
 
+void QtFirebaseAdMobBase::onPresentationStateChanged(int state)
+{
```

Comment:
PresentationStateChanges should be kept separate in each AdMob type  (banner, interstitial, Rewarded etc.)
Having e.g. `PresentationStateVideoHasStarted` ENUMs available to banners is confusing. Also Google is adding and removing features from time to time so having glorious base classes makes it harder to maintain changes in the derived classes.

Plus the presentation state cycle can be different for each Ad type.